### PR TITLE
build(ci): add bot to print binary size diff compared to last release

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

**Comment by efekarakus** [comment]
I want to do a slight tweak where the bot updates an existing comment instead of posting a new one for every time the PR changes

**Comment by efekarakus** [comment]
Idea for tomorrow: if the diff for increase in binary size is larger than 10%, then it will apply the "do-not-merge" label so that merging the PR should be acknowledge and an on purpose move by the author

File: .github/workflows/ci_writer.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
+name: ci annotate
+
+on:
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - reopened
```
Comment (dannyrandall) [inline .github/workflows/ci_writer.yml]: does reopened potentially contain new code?🤔

File: .github/workflows/ci_writer.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                }
+            
+                let symbol = "";
+                if (delta > 5) {
+                  symbol = "😭 +";
+                } else if (delta > 1) {
+                  symbol = "🥺 +"
+                } else if (delta > 0) {
+                  symbol = "+";
+                } else if (delta < 0) {
```
Comment (dannyrandall) [inline .github/workflows/ci_writer.yml]:
```suggestion
                } else if (delta <= 0) {
```

Comment (dannyrandall) [reply .github/workflows/ci_writer.yml]: or change default symbol on 115

File: .github/workflows/ci_writer.yml [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+          make release
+          echo "MAC_AMD_KIB=$(du -k ./bin/local/copilot-darwin-amd64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "MAC_ARM_KIB=$(du -k ./bin/local/copilot-darwin-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_AMD_KIB=$(du -k ./bin/local/copilot-linux-amd64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "LINUX_ARM_KIB=$(du -k ./bin/local/copilot-linux-arm64 | awk '{ print $1}')" >> $GITHUB_OUTPUT
+          echo "WINDOWS_AMD_KIB=$(du -k ./bin/local/copilot.exe | awk '{ print $1}')" >> $GITHUB_OUTPUT
+
+      - name: Calculate size change
+        id: diff
+        uses: actions/github-script@v6
```
Comment (dannyrandall) [inline .github/workflows/ci_writer.yml:60]: wow this is cool!

File: .github/workflows/ci_writer.yml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: ${{steps.diff.outputs.result}}
+
+      - name: Update comment if it exists
+        if: steps.fc.outputs.comment-id != ''
+        uses: peter-evans/create-or-update-comment@v2
+        with:
+          comment-id: ${{ steps.fc.outputs.comment-id }}
+          body: ${{steps.diff.outputs.result}}
```
Comment (dannyrandall) [inline .github/workflows/ci_writer.yml]:
```suggestion
      - name: Create or update comment
        uses: peter-evans/create-or-update-comment@v2
        with:
          comment-id: ${{ steps.fc.outputs.comment-id }}
          issue-number: ${{ github.event.pull_request.number }}
          body: ${{steps.diff.outputs.result}}
```
from https://github.com/peter-evans/create-or-update-comment#where-to-find-the-id-of-a-comment

Comment (dannyrandall) [reply .github/workflows/ci_writer.yml]: oh also maybe need to set
`edit-mode: replace`

Comment (efekarakus) [reply .github/workflows/ci_writer.yml]: Thanks for the great feedback, I addressed them all!

**Review by dannyrandall — APPROVED** [review]
awesome!!!

**Review by iamhopaul123 — APPROVED** [review]

**Pull request merged**